integration/tykoas_test.go

```diff
@@ -0,0 +1,221 @@
+package integration
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/TykTechnologies/tyk-operator/api/v1alpha1"
+	"github.com/TykTechnologies/tyk-operator/controllers"
+	tykClient "github.com/TykTechnologies/tyk-operator/pkg/client"
+	"github.com/TykTechnologies/tyk-operator/pkg/client/klient"
+	"github.com/TykTechnologies/tyk-operator/pkg/environment"
+	"github.com/buger/jsonparser"
+	"github.com/matryer/is"
+	"golang.org/x/mod/semver"
+	v1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	cr "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestOASCreate(t *testing.T) {
+	var (
+		testNS string
+		tykOAS *v1alpha1.TykOAS
+		eval   = is.New(t)
+		tykCtx context.Context
+	)
+
+	f := features.New("Create Tyk OAS API").
+		Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			var err error
+
+			// Obtain Environment configuration to be able to connect Tyk.
+			tykEnv, err := generateEnvConfig(ctx, c)
+			eval.NoErr(err)
+
+			res := semver.Compare(tykEnv.TykVersion, "v4.1")
+			if res < 0 {
+				t.Skip("OAS support is added in Tyk in v4.1")
+			}
+
+			tykCtx = tykClient.SetContext(context.Background(), tykClient.Context{
+				Env: tykEnv,
+				Log: log.NullLogger{},
+			})
+
+			var ok bool
+
+			testNS, ok = ctx.Value(ctxNSKey).(string)
+			eval.True(ok)
+
+			tykOAS, _, err = createTestOASApi(ctx, testNS, c)
+			eval.NoErr(err)
+
+			err = waitForTykResourceCreation(c, tykOAS)
+			eval.NoErr(err)
+
+			return ctx
+		}).Assess("Ensure OAS definition is created on Tyk",
+		func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+			exists := klient.Universal.TykOAS().Exists(tykCtx, tykOAS.Status.ID)
+			eval.True(exists)
```

Comment:
Maybe we can retry if `exists` variable does not match desired outcome? What do you think? Intermittent connection issues between controller and Tyk might failure on this test case even OAS has been created on Tyk

Reply:
Resolved in [66cb5ce](https://github.com/TykTechnologies/tyk-operator/pull/687/commits/66cb5cef364d99ac5601a241c0a21b8e4242dde0)